src/components/Navbar.js

```diff
@@ -1,101 +1,81 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "gatsby";
 import github from "../img/github-icon.svg";
 import logo from "../img/logo.svg";
 
-const Navbar = class extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      active: false,
-      navBarActiveClass: "",
-    };
-  }
+const Navbar = () => {
+  const [isActive, setIsActive] = useState(false);
 
-  toggleHamburger() {
-    // toggle the active boolean in the state
-    this.setState(
-      {
-        active: !this.state.active,
-      },
-      // after state has been updated,
-      () => {
-        // set the class in state for the navbar accordingly
-        this.state.active
-          ? this.setState({
-              navBarActiveClass: "is-active",
-            })
-          : this.setState({
-              navBarActiveClass: "",
-            });
-      }
-    );
-  }
-
-  render() {
-    return (
-      <nav
-        className="navbar is-transparent"
-        role="navigation"
-        aria-label="main-navigation"
-      >
-        <div className="container">
-          <div className="navbar-brand">
-            <Link to="/" className="navbar-item" title="Logo">
-              <img src={logo} alt="Kaldi" style={{ width: "88px" }} />
-            </Link>
-            {/* Hamburger menu */}
-            <div
-              className={`navbar-burger burger ${this.state.navBarActiveClass}`}
-              data-target="navMenu"
-              role="menuitem"
-              tabIndex={0}
-              onKeyPress={() => this.toggleHamburger()}
-              onClick={() => this.toggleHamburger()}
-            >
-              <span />
-              <span />
-              <span />
-            </div>
-          </div>
-          <div
-            id="navMenu"
-            className={`navbar-menu ${this.state.navBarActiveClass}`}
+  return (
+    <nav
+      className="navbar is-transparent"
+      role="navigation"
+      aria-label="main-navigation"
+    >
+      <div className="container">
+        <div className="navbar-brand">
+          <Link to="/" className="navbar-item" title="Logo">
+            <img src={logo} alt="Kaldi" style={{ width: "88px" }} />
+          </Link>
+          {/* Hamburger menu */}
+          <button
+            className={`navbar-burger burger ${isActive && "is-active"}`}
+            aria-expanded={isActive}
+            tabIndex={0}
+            onKeyPress={() => setIsActive(!isActive)}
```

Comment:
Because we're using a button we're okay to remove the tabIndex. We'll be able to still focus it without any worry.

Also because we're using onClick, we don't need to use onKeyPress because if we have both it might end up opening and then shortly closing it because I think keypress vs. onClick happen in one after the other.

Reply:
🤦🏾‍♀️  you are 100% correct!